db/migrate/20201211160500_add_creator_user_to_notes.rb

```diff
@@ -0,0 +1,6 @@
+class AddCreatorUserToNotes < ActiveRecord::Migration[6.0]
+  def change
+    Note.connection.execute('DELETE FROM notes')
+    add_reference :notes, :creator_user, null: false, foreign_key: { to_table: :users }
```

Comment:
My local linter adds the following warning here: `convention:Rails/NotNullColumn` Is this a warning we should care about?

Reply:
Here are some example from the [documentation](https://rubocop.readthedocs.io/projects/rails/en/stable/cops_rails/#railsnotnullcolumn), providing a rationale for this rule:

```Ruby
# bad
add_column :users, :name, :string, null: false
add_reference :products, :category, null: false

# good
add_column :users, :name, :string, null: true
add_column :users, :name, :string, null: false, default: ''
add_reference :products, :category
add_reference :products, :category, null: false, default: 1
```

Reply:
@chrisma Hi, right now we just are just saving the user_id of the current_user in the notes model. For that we have the attribute `creator_user_id`. So how can you add the support for `@note.creator = current_user` and why is that necessary?

Reply:
Okay, already done